pkg/sql/plan/function/operator/compare.go

```diff
@@ -182,43 +182,50 @@ func CompareBytesNe(v1, v2 []byte, s1, s2 int32) bool {
 
 func CompareString(vs []*vector.Vector, fn compStringFn, proc *process.Process) (*vector.Vector, error) {
 	v1, v2 := vs[0], vs[1]
-	col1, col2 := vector.MustBytesCols(v1), vector.MustBytesCols(v2)
+
 	if v1.IsScalarNull() || v2.IsScalarNull() {
 		return handleScalarNull(v1, v2, proc)
 	}
 
 	if v1.IsScalar() && v2.IsScalar() {
+		col1, col2 := vector.MustBytesCols(v1), vector.MustBytesCols(v2)
```

Comment:
I will not use MustByteCols at all.   In this case, the pseudo code should look like this,
if v1.isScalar 
    varlen1 = v1.MustTCol
    result = v2.EqualValena(varlen1, area)
    // where v2.EuqalVarlena is implemented as 
    for i in range(1000):
         result[i] = varlenaEq(varlen2[i], varlen1, area))   // this varelanEq function should be 3 uint64 ==